packages/react/src/components/ComposedModal/ComposedModal.tsx

```diff
@@ -0,0 +1,457 @@
+            focusContainerElement.querySelector(selectorPrimaryFocus);
+          if (primaryFocusElement) {
+            primaryFocusElement.focus();
+            return;
+          }
+          if (button.current) {
+            button.current.focus();
+          }
+        }
+      };
+
+      if (!open) {
+        return;
+      }
+
+      if (innerModal.current) {
+        focusButton(innerModal.current);
+      }
+    }, [open, selectorPrimaryFocus]);
+
+    return (
+      <div
+        {...rest}
+        role="presentation"
+        ref={ref}
+        aria-hidden={!open}
+        onBlur={handleBlur}
+        onMouseDown={handleMousedown}
+        onKeyDown={handleKeyDown}
+        className={modalClass}>
+        <div
+          className={containerClass}
+          role="dialog"
+          aria-modal="true"
+          aria-label={ariaLabel ? ariaLabel : generatedAriaLabel}
+          aria-labelledby={ariaLabelledBy}>
+          {/* Non-translatable: Focus-wrap code makes this `<button>` not actually read by screen readers */}
+          <button
+            type="button"
+            ref={startSentinel}
+            className={`${prefix}--visually-hidden`}>
+            Focus sentinel
+          </button>
+          <div ref={innerModal} className={`${prefix}--modal-container-body`}>
+            {childrenWithProps}
+          </div>
+          {/* Non-translatable: Focus-wrap code makes this `<button>` not actually read by screen readers */}
+          <button
+            type="button"
+            ref={endSentinel}
+            className={`${prefix}--visually-hidden`}>
+            Focus sentinel
+          </button>
+        </div>
+      </div>
+    );
+  }
+);
+
+ComposedModal.propTypes = {
+  /**
+   * Specify the aria-label for bx--modal-container
+   */
+  ['aria-label']: PropTypes.string,
+
+  /**
+   * Specify the aria-labelledby for bx--modal-container
+   */
+  ['aria-labelledby']: PropTypes.string,
+
+  /**
+   * Specify the content to be placed in the ComposedModal
+   */
+  children: PropTypes.node,
+
+  /**
+   * Specify an optional className to be applied to the modal root node
+   */
+  className: PropTypes.string,
+
+  /**
+   * Specify an optional className to be applied to the modal node
+   */
+  containerClassName: PropTypes.string,
+
+  /**
+   * Specify whether the primary button should be replaced with danger button.
+   * Note that this prop is not applied if you render primary/danger button by yourself
+   */
+  danger: PropTypes.bool,
+
+  /**
+   * Specify whether the Modal content should have any inner padding.
+   */
+  isFullWidth: PropTypes.bool,
+
+  /**
+   * Specify an optional handler for closing modal.
+   * Returning `false` here prevents closing modal.
+   */
+  onClose: PropTypes.func,
+
+  /**
+   * Specify an optional handler for the `onKeyDown` event. Called for all
+   * `onKeyDown` events that do not close the modal
+   */
+  // @ts-expect-error: Invalid derived type
```

Comment:
let's try to prevent using these as much as possible, what's the error this is generating here?